test/langtools/tools/javac/enum/T7176515.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 7176515
+ * @summary ExceptionInInitializerError for an enum with multiple switch statements
+ */
+
+import java.math.RoundingMode;
+
+public class T7176515 {
+
+    public enum MyEnum {
+        FIRST(RoundingMode.CEILING),
+        SECOND(RoundingMode.HALF_DOWN),
+        THIRD(RoundingMode.UNNECESSARY),
+        FOURTH(RoundingMode.HALF_EVEN),
+        FIFTH(RoundingMode.HALF_DOWN),
+        SIXTH(RoundingMode.CEILING),
+        SEVENTH(RoundingMode.UNNECESSARY);
+
+        private final RoundingMode mode;
+
+        private MyEnum(RoundingMode mode) {
+            switch (mode) {
+            case CEILING:
+            case HALF_DOWN:
+            case UNNECESSARY:
+            case HALF_EVEN:
+                break;
+            default:
+                throw new IllegalArgumentException();
+            }
+            this.mode = mode;
+        }
+
+        public boolean isOdd() {
+            switch (this) {
+            case FIRST:
+            case THIRD:
+            case FIFTH:
+            case SEVENTH:
+                return true;
+            default:
+                return false;
+            }
+        }
+    }
+
+    public enum Nested {
+        AAA(MyEnum.FIRST),
+        BBB(MyEnum.THIRD),
+        CCC(MyEnum.FIFTH),
+        DDD(MyEnum.SEVENTH),
+        EEE(MyEnum.SECOND);
+
+        private Nested(MyEnum x) {
+// Bug still occurs if this is uncommented
+//            switch (x) {
+//            default:
+//                break;
+//            }
+        }
+    }
+
+    public static void main(String[] args) {
+        boolean shouldBeOdd = true;
+        for (MyEnum x : MyEnum.values()) {
+            assert x.isOdd() == shouldBeOdd;
```

Comment:
While we do typically run the tests with assertions enabled, it is preferable if the tests did not rely on that for proper operation.

Reply:
Good idea - thanks. Fixed in 8cd5b49c5f2.